values.go

```diff
@@ -1552,12 +1565,18 @@ func (v Array) WithType(arrayType ArrayType) Array {
 }
 
 func (v Array) ToGoValue() any {
-	ret := make([]any, len(v.Values))
+	ret := []any{}
 
-	for i, e := range v.Values {
-		ret[i] = e.ToGoValue()
+	for _, e := range v.Values {
+		value := e.ToGoValue()
+		if value != nil {
+			ret = append(ret, value)
+		}
 	}
 
+	if len(ret) == 0 {
+		return nil
+	}
```

Comment:
Maybe move this to the top of the function, before unnecessarily allocating an empty slice

Reply:
If we want to skip nil values in the list we cant do this. Do we want to do that or not? I personally prefer it, but if you do not want it I can change it around.